lib/pushmi_pullyu/cli.rb

```diff
@@ -160,15 +130,54 @@ def run_tick_loop
     end
   end
 
+  def running?
+    @running
+  end
+
+  def setup_log
+    if config.daemonize
+      PushmiPullyu::Logging.initialize_logger(config.logfile)
+    else
+      logger.formatter = PushmiPullyu::Logging::SimpleFormatter.new
+    end
+    logger.level = ::Logger::DEBUG if config.debug
+  end
+
+  def setup_signal_traps
+    Signal.trap('INT') { Thread.new { shutdown } }
+    Signal.trap('TERM') { Thread.new { shutdown } }
+    Signal.trap('HUP') { Thread.new { PushmiPullyu::Logging.reopen } }
+  end
+
+  def setup_queue
+    @queue = PushmiPullyu::PreservationQueue.new(connection: { host: config.redis_host, port: config.redis_port },
+                                                 queue_name: config.queue_name,
+                                                 age_at_least: config.minimum_age)
+  end
+
+  def shutdown
+    exit!(1) unless running?
```

Comment:
Original PR comment got squashed here: https://github.com/ualbertalib/pushmi_pullyu/pull/44#discussion_r114667295

Yeah comment is worth having though!

Reason for exit!(1)? Since you force shutting it down I figure that's not a normal successful program termination. But we can change that to exit 0 if you think that's better